### PR TITLE
Mockgen import cycle not allowed && recover missing mockgens

### PR DESCRIPTION
**Pull request by adriacidre**
`make mock` generates an import cycle error

```
~/go/src/github.com/status-im/status-go$ make setup
can't load package: import cycle not allowed
package github.com/status-im/status-go/geth/transactions/fake
        imports github.com/status-im/status-go/geth/transactions/fake
go get -u github.com/alecthomas/gometalinter
```

Important changes:
- [x] Changed `mockgen` mode to use source instead of package based generation on transactions package
- [x] Added accounts mock generation
- [x] Added jail mock generation
- [x] Committed generated mocks to avoid unwished conflicts

Closes #

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -112,9 +112,11 @@ mock-install: ##@other Install mocking tools
...
-	mockgen -package=fcm         -destination=geth/notifications/push/fcm/client_mock.go github.com/status-im/status-go/geth/notifications/push/fcm FirebaseClient,Notifier
-	mockgen -package=fake        -destination=geth/transactions/fake/mock.go             github.com/status-im/status-go/geth/transactions/fake PublicTransactionPoolAPI
+	mockgen -package=mailservice  -destination=geth/mailservice/mailservice_mock.go       github.com/status-im/status-go/geth/mailservice ServiceProvider
+	mockgen -package=fcm          -destination=geth/notifications/push/fcm/client_mock.go github.com/status-im/status-go/geth/notifications/push/fcm FirebaseClient,Notifier
+	mockgen -package=fake         -destination=geth/transactions/fake/mock.go             -source=geth/transactions/fake/txservice.go PublicTransactionPoolAPI
+	mockgen -package=account      -destination=geth/account/accounts_mock.go              -source=geth/account/accounts.go
+	mockgen -package=jail         -destination=geth/jail/cell_mock.go                  		-source=geth/jail/cell.go
```
Comment (themue) [inline Makefile]: Looks like indented one tab to far.

**Review by themue — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (themue) [reply Makefile]: Meaning `-source`.

**Review by mandrigin — APPROVED** [review]
LGTM!

**Comment by adambabik** [comment]
Please, run the build in Jenkins with this branch to make sure it passes there as well. We had some issues with mockgen and source mode in Jenkins.

**Comment by adriacidre** [comment]
@adambabik so [travis build](https://jenkins.status.im/job/status-go/job/status-go-manual/177/) passed after rebasing with develop. I'm going to merge this PR

**Pull request merged**